examples/BeliefPropagation/BeliefPropagationFunctions.jl

```diff
@@ -0,0 +1,170 @@
+using ITensors
+using ITensorNetworks
+using NamedGraphs
+using DataGraphs
+using MultiDimDictionaries
+using Graphs
+using Random
+using LinearAlgebra
+
+maybe_only(x) = x
+maybe_only(x::Tuple{T}) where {T} = only(x)
+
+#Function to construct the initial message tensors associated with a given ItensorNetwork (two for each edge in the graph, one forward one backward)
+function construct_initial_mts(psi::ITensorNetwork, g::NamedDimGraph, s::IndsNetwork)
+    #Make empty lists
+    forwardmts = ITensor[]
+    backwardmts = ITensor[]
```

Comment:
What about storing the message tensors in a `Dictionary` that maps the edges the message tensors are on to the message tensor? Then it is easy to look up the message tensors associated with a vertex by searching through the incident edges of the vertex.